AWSDDBMetadataAdapterTest::adaptTest is flaky

## This is a...

<pre><code>
[ ] Feature request
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  <!-- Please search GitHub for a similar issue or PR before submitting -->
[ ] Documentation issue or request
</code></pre>

## Description
Fails intermittently with:

```
[ERROR] Tests run: 2, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 2.236 s <<< FAILURE! - in io.syndesis.connector.aws.ddb.metadata.AWSDDBMetadataAdapterTest
[ERROR] adaptTest(io.syndesis.connector.aws.ddb.metadata.AWSDDBMetadataAdapterTest)  Time elapsed: 2.053 s  <<< FAILURE!
java.lang.AssertionError: 

Expected: properties
     but none found

	at io.syndesis.connector.aws.ddb.metadata.AWSDDBMetadataAdapterTest.adaptTest(AWSDDBMetadataAdapterTest.java:97)
```

--- a/app/connector/aws-ddb/src/main/java/io/syndesis/connector/aws/ddb/metadata/AWSDDBMetadataRetrieval.java
+++ b/app/connector/aws-ddb/src/main/java/io/syndesis/connector/aws/ddb/metadata/AWSDDBMetadataRetrieval.java
@@ -53,9 +53,10 @@ import org.slf4j.LoggerFactory;
 /**
  * Extract all automatic info (column names) to enrich the properties.
  */
-public final class AWSDDBMetadataRetrieval extends ComponentMetadataRetrieval {
+public class AWSDDBMetadataRetrieval extends ComponentMetadataRetrieval {
 
     private static final Logger LOG = LoggerFactory.getLogger(AWSDDBMetadataRetrieval.class);
+
     @Override
     protected MetaDataExtension resolveMetaDataExtension(CamelContext context,
                                                          Class<? extends MetaDataExtension> metaDataExtensionClass,
@@ -64,7 +65,6 @@ public final class AWSDDBMetadataRetrieval extends ComponentMetadataRetrieval {
         return new AWSDDBConnectorMetaDataExtension(context);
     }
 
-
     @Override
     protected SyndesisMetadata adapt(CamelContext context, String componentId, String actionId,
                                      Map<String, Object> properties, MetaData metadata) {
@@ -72,21 +72,11 @@ public final class AWSDDBMetadataRetrieval extends ComponentMetadataRetrieval {
         return adaptForDDB(properties, setupDefaultValues(properties));
     }
 
-
-
     protected Map<String, List<PropertyPair>> setupDefaultValues(Map<String, Object> properties) {
         Map<String, List<PropertyPair>> res = new HashMap<String, List<PropertyPair>>();
 
         try {
-            AWSCredentials credentials = new BasicAWSCredentials(properties.get("accessKey").toString(),
-                properties.get("secretKey").toString());
-            AWSCredentialsProvider credentialsProvider = new AWSStaticCredentialsProvider(credentials);
-            AmazonDynamoDB client =
-                AmazonDynamoDBClientBuilder.standard().withCredentials(credentialsProvider)
-                    .withRegion(Regions.valueOf(properties.get("region").toString())).build();
-
-            DescribeTableResult table = client.describeTable(properties.get(
-                "tableName").toString());
+            DescribeTableResult table = fetchTableDescription(properties);
 
             StringBuilder element = new StringBuilder("{");
             StringBuilder attributes = new StringBuilder();
@@ -119,10 +109,20 @@ public final class AWSDDBMetadataRetrieval extends ComponentMetadataRetrieval {
             res.put("attributes", list);
 
         } catch (AmazonDynamoDBException t) {
-           LOG.error("Couldn't connect to Amazon services. No suggestions on the fields.", t);
+            LOG.error("Couldn't connect to Amazon services. No suggestions on the fields.", t);
         }
 
         return res;
+    }
+
+    DescribeTableResult fetchTableDescription(Map<String, Object> properties) {
+        AWSCredentials credentials = new BasicAWSCredentials(properties.get("accessKey").toString(),
+            properties.get("secretKey").toString());
+        AWSCredentialsProvider credentialsProvider = new AWSStaticCredentialsProvider(credentials);
+        AmazonDynamoDB client = AmazonDynamoDBClientBuilder.standard().withCredentials(credentialsProvider)
+            .withRegion(Regions.valueOf(properties.get("region").toString())).build();
+
+        return client.describeTable(properties.get("tableName").toString());
     }
 
     /**
@@ -131,8 +131,8 @@ public final class AWSDDBMetadataRetrieval extends ComponentMetadataRetrieval {
      * @param properties
      * @return
      */
-    private SyndesisMetadata adaptForDDB(final Map<String, Object> properties,
-                                         Map<String, List<PropertyPair>> enrichedProperties) {
+    private static SyndesisMetadata adaptForDDB(final Map<String, Object> properties,
+        Map<String, List<PropertyPair>> enrichedProperties) {
 
         Map<String, AttributeValue> element = Util.getAttributeValueMap("element", properties);
 
@@ -145,8 +145,8 @@ public final class AWSDDBMetadataRetrieval extends ComponentMetadataRetrieval {
             attributes = splitter.splitToList(optionAttributes);
         }
 
-        //fallback to use the list of attributes on the filter
-        //this is used always on put-item
+        // fallback to use the list of attributes on the filter
+        // this is used always on put-item
         if (attributes.isEmpty()) {
             attributes.addAll(element.keySet());
         }

--- a/app/connector/aws-ddb/src/test/java/io/syndesis/connector/aws/ddb/metadata/AWSDDBMetadataAdapterTest.java
+++ b/app/connector/aws-ddb/src/test/java/io/syndesis/connector/aws/ddb/metadata/AWSDDBMetadataAdapterTest.java
@@ -15,27 +15,29 @@
  */
 package io.syndesis.connector.aws.ddb.metadata;
 
+import java.util.Collections;
 import java.util.HashMap;
 import java.util.Map;
 import java.util.Optional;
 
-import static org.junit.Assert.assertEquals;
-
-import com.fasterxml.jackson.core.JsonProcessingException;
-import com.fasterxml.jackson.databind.ObjectWriter;
-import io.swagger.models.Swagger;
 import io.syndesis.common.util.Json;
-import io.syndesis.common.util.openapi.OpenApiHelper;
 import io.syndesis.connector.aws.ddb.AWSDDBConfiguration;
 import io.syndesis.connector.support.verifier.api.SyndesisMetadata;
+
 import org.apache.camel.CamelContext;
 import org.apache.camel.component.extension.MetaDataExtension.MetaData;
 import org.apache.camel.impl.DefaultCamelContext;
 import org.json.JSONException;
-import org.junit.Ignore;
 import org.junit.Test;
 import org.skyscreamer.jsonassert.JSONAssert;
 import org.skyscreamer.jsonassert.JSONCompareMode;
+
+import com.amazonaws.services.dynamodbv2.model.AttributeDefinition;
+import com.amazonaws.services.dynamodbv2.model.DescribeTableResult;
+import com.amazonaws.services.dynamodbv2.model.ScalarAttributeType;
+import com.amazonaws.services.dynamodbv2.model.TableDescription;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectWriter;
 
 public class AWSDDBMetadataAdapterTest {
 
@@ -51,7 +53,18 @@ public class AWSDDBMetadataAdapterTest {
         parameters.put(AWSDDBConfiguration.ACCESSKEY, AWSDDBConfiguration.ACCESSKEY_VALUE);
         parameters.put(AWSDDBConfiguration.TABLENAME, AWSDDBConfiguration.TABLENAME_VALUE);
         Optional<MetaData> metadata = ext.meta(parameters);
-        AWSDDBMetadataRetrieval adapter = new AWSDDBMetadataRetrieval();
+        AWSDDBMetadataRetrieval adapter = new AWSDDBMetadataRetrieval() {
+            @Override
+            DescribeTableResult fetchTableDescription(Map<String, Object> properties) {
+                DescribeTableResult tableDescriptionResult = new DescribeTableResult();
+                TableDescription tableDescription = new TableDescription();
+                tableDescription.setAttributeDefinitions(Collections.singletonList(new AttributeDefinition("clave", ScalarAttributeType.S)));
+
+                tableDescriptionResult.setTable(tableDescription);
+
+                return tableDescriptionResult;
+            }
+        };
 
         ObjectWriter writer = Json.writer();
         SyndesisMetadata syndesisMetaData2 = adapter.adapt(camelContext,
@@ -109,7 +122,18 @@ public class AWSDDBMetadataAdapterTest {
         parameters.put("accessKey", "invalidKey");
         parameters.put("tableName", "TestTable");
         Optional<MetaData> metadata = ext.meta(parameters);
-        AWSDDBMetadataRetrieval adapter = new AWSDDBMetadataRetrieval();
+        AWSDDBMetadataRetrieval adapter = new AWSDDBMetadataRetrieval() {
+            @Override
+            DescribeTableResult fetchTableDescription(Map<String, Object> properties) {
+                DescribeTableResult tableDescriptionResult = new DescribeTableResult();
+                TableDescription tableDescription = new TableDescription();
+                tableDescription.setAttributeDefinitions(Collections.emptyList());
+
+                tableDescriptionResult.setTable(tableDescription);
+
+                return tableDescriptionResult;
+            }
+        };
 
         ObjectWriter writer = Json.writer();
         SyndesisMetadata syndesisMetaData2 = adapter.adapt(camelContext,
@@ -138,6 +162,13 @@ public class AWSDDBMetadataAdapterTest {
             "    \"metadata\" : {\n" +
             "      \"variant\" : \"collection\"\n" +
             "    }\n" +
+            "  },\n" +
+            "  \"properties\" : {\n" +
+            "    \"attributes\" : [ { } ],\n" +
+            "    \"element\" : [ {\n" +
+            "      \"displayValue\" : \"{}\",\n" +
+            "      \"value\" : \"{}\"\n" +
+            "    } ]\n" +
             "  }\n" +
             "}\n";
 

